vignettes/migrating.Rmd

```diff
@@ -0,0 +1,110 @@
+
+The most obvious user-facing change is that there is (almost) no [YAML](https://en.wikipedia.org/wiki/YAML), with the definition of inputs and outputs for a report now defined within an R script (`orderly.R`).  So an orderly report that previously had an `orderly.yml` file that looked like
+
+```yaml
+parameters:
+  n_min:
+    default: 10
+script: script.R
+source:
+  - functions.R
+resources:
+  - metadata.csv
+depends:
+  raw_data:
+    id: latest
+    use:
+      raw_data.csv: data.csv
+artefacts:
+  data:
+    description: Processed data
+    filenames: data.rds
+```
+
+would end up within an `orderly.R` script that looks like:
+
+```r
+orderly2::orderly_parameters(n_min = 10)
+orderly2::orderly_dependency("raw_data", "latest", 
+                             use = c("raw_data.csv" = "data.csv"))
+orderly2::orderly_resource("metadata.csv")
+orderly2::orderly_artefact("Processed data", "data.rds"))
+source("functions.R")
+```
+
+We think this is much clearer, and comes with documentation and autocomplete support in most IDEs.
+
+In fact, for simple reports, **no** special functions are required, though you'll find that some will be useful (see `vignette("orderly2")`)
+
+### Implications
+
+This change has widespread implications:
+
+* you can program against things like dependencies, creating a `for` loop over a series of parameter values, or conditionally depending on other reports
+* any R script can be the basis of an orderly report, and you can add orderly functions annotating inputs and outputs progressively
+
+## Database support has been moved into a plugin
+
+In version 1, we had built-in support for accessing data from SQL databases; this has moved within the [`orderly.db`](https://github.com/mrc-ide/orderly.db) plugin. All major features are supported.
+
+## No more commit
+
+`orderly2` no longer requires a separate `orderly_commit()` call after `orderly_run()`; we no longer make a distinction between local draft and archive packets. Instead, we have added finer-grained control over where dependencies are resolved from (locally, or from some subset of your servers), which generalises the way that draft/archive was used in practice. See the "Dependency resolution" section of `?outpack_packet_use_dependency` (itself used from `orderly2::orderly_dependency`) for more details.
+
+## No more testing or development mode
+
+We have had two different, but unsatisfactory, mechanisms for developing an orderly report:
+
+* `orderly_test_start` (up to version 1.1.0)
+* `orderly_develop_start` (from 1.1.0 onwards)
+
+These worked by doing the initial setup, and copying of dependencies etc to a location you could work (a new draft for `orderly_test_start` and the source directoryt for `orderly_develop_start`). From `orderly2` you can just work directly within the source directory, and so long as your working directory is set to `src/<report-name>`, all `orderly2` commands will work as expected.
```

Comment:
do people need to be extra careful on not committing outputs (orderly_develop_clean was great). also typo in directory